node/src/common/AsyncQueue.h

```diff
@@ -0,0 +1,252 @@
+//
+// Copyright (c) 2015, Microsoft Corporation
+//
+// Permission to use, copy, modify, and/or distribute this software for any
+// purpose with or without fee is hereby granted, provided that the above
+// copyright notice and this permission notice appear in all copies.
+//
+// THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+// WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+// MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY
+// SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+// WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+// ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF OR
+// IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+//
+
+namespace OpenT2T
+{
+
+// Interface implemented by callers
+template <class QueueItem>
+class IQueueItemHandler
+{
+public:
+    virtual void OnStarted() = 0;
+    virtual void OnProcessQueueItem(QueueItem& item) = 0;
+    virtual void OnStopped() = 0;
+};
+
+static bool s_crtIsTerminating = false;
```

Comment:
I copied AsyncQueue.h [from the bridge repo](https://github.com/tjaffri/msiot-samples/blob/develop/AllJoyn/Platform/Bridge/ScriptAdapterLib/AsyncQueue.h), so I didn't write this code. But I agree that's less than ideal. I'll see if some of the code be put into a .cpp file instead.